app/services/katello/pulp3/generic_content_unit.rb

```diff
@@ -0,0 +1,51 @@
+module Katello
+  module Pulp3
+    class GenericContentUnit < PulpContentUnit
+      include LazyAccessor
+      CONTENT_TYPE = "generic_content_unit".freeze
+
+      def self.pulp_units_batch_for_repo(repository, options = {})
+        fetch_identifiers = options.fetch(:fetch_identifiers, false)
+        page_size = options.fetch(:page_size, Setting[:bulk_load_size])
+        repository_version_href = repository.version_href
+        page_opts = { "offset" => 0, repository_version: repository_version_href, limit: page_size }
+        page_opts[:fields] = self.const_get(:PULP_INDEXED_FIELDS).join(",") if self.constants.include?(:PULP_INDEXED_FIELDS)
+        page_opts[:fields] = 'pulp_href' if fetch_identifiers
+        response = {}
+        Enumerator.new do |yielder|
+          loop do
+            page_opts = page_opts.with_indifferent_access
+            break unless (
+              (response["count"] && page_opts["offset"] < response["count"]) ||
+                page_opts["offset"] == 0)
+            page_opts = page_options page_opts
+            response = fetch_content_list page_opts, repository.repository_type
```

Comment:
Same comment here as above. You should use `pulp_units_batch_for_repo` that's in `PulpContentUnit` instead of redefining the method, since 99% of the code is the same.  You can add a conditional check there to make it work.